Sources/_StringProcessing/ConsumerInterface.swift

```diff
@@ -0,0 +1,602 @@
+    case .bidiControl:
+      break
+
+
+    case .bidiMirrored: 
+      return consumeScalarProp(\.isBidiMirrored)
+    case .cased:
+      return consumeScalarProp(\.isCased)
+    case .compositionExclusion:
+      break
+    case .caseIgnorable:
+      return consumeScalarProp(\.isCaseIgnorable)
+    case .changesWhenCasefolded:
+      return consumeScalarProp(\.changesWhenCaseFolded)
+    case .changesWhenCasemapped:
+      return consumeScalarProp(\.changesWhenCaseMapped)
+    case .changesWhenNFKCCasefolded:
+      return consumeScalarProp(\.changesWhenNFKCCaseFolded)
+    case .changesWhenLowercased:
+      return consumeScalarProp(\.changesWhenLowercased)
+    case .changesWhenTitlecased:
+      return consumeScalarProp(\.changesWhenTitlecased)
+    case .changesWhenUppercased:
+      return consumeScalarProp(\.changesWhenUppercased)
+    case .dash:
+      return consumeScalarProp(\.isDash)
+    case .deprecated:
+      return consumeScalarProp(\.isDeprecated)
+    case .defaultIgnorableCodePoint:
+      return consumeScalarProp(\.isDefaultIgnorableCodePoint)
+    case .diacratic: // spelling?
+      return consumeScalarProp(\.isDiacritic)
+    case .emojiModifierBase:
+      break // availability
+    case .emojiComponent:
+      break
+    case .emojiModifier:
+      break // availability
+    case .emoji:
+      break // availability
+    case .emojiPresentation:
+      break // availability
+    case .extender:
+      return consumeScalarProp(\.isExtender)
+    case .extendedPictographic:
+      break
+    case .fullCompositionExclusion:
+      return consumeScalarProp(\.isFullCompositionExclusion)
+    case .graphemeBase:
+      return consumeScalarProp(\.isGraphemeBase)
+    case .graphemeExtended:
+      break
+    case .graphemeLink:
+      break
+    case .hexDigit:
+      return consumeScalarProp(\.isHexDigit)
+    case .hyphen:
+      break
+    case .idContinue:
+      return consumeScalarProp(\.isIDContinue)
+    case .ideographic:
+      return consumeScalarProp(\.isIdeographic)
+    case .idStart:
+      return consumeScalarProp(\.isIDStart)
+    case .idsBinaryOperator:
+      return consumeScalarProp(\.isIDSBinaryOperator)
+    case .idsTrinaryOperator:
+      return consumeScalarProp(\.isIDSTrinaryOperator)
+    case .joinControl:
+      return consumeScalarProp(\.isJoinControl)
+    case .logicalOrderException:
+      return consumeScalarProp(\.isLogicalOrderException)
+    case .lowercase:
+      return consumeScalarProp(\.isLowercase)
+    case .math:
+      return consumeScalarProp(\.isMath)
+    case .noncharacterCodePoint:
+      return consumeScalarProp(\.isNoncharacterCodePoint)
+    case .otherAlphabetic:
+      break
+    case .otherDefaultIgnorableCodePoint:
+      break
+    case .otherGraphemeExtended:
+      break
+    case .otherIDContinue:
+      break
+    case .otherIDStart:
+      break
+    case .otherLowercase:
+      break
+    case .otherMath:
+      break
+    case .otherUppercase:
+      break
```

Comment:
These properties are unfortunate because they get scooped up with their normal counterpart (`Alphabetic`) to match ICU's behavior when asking if a scalar `isAlphabetic`. If we need to make the distinction here, then that'll probably need to be a separate table.

Reply:
Hmm, IIRC they were because Unicode historically used general categories for these kind of queries, but it became apparent that a universal categorization is not great for this use. IIRC `isAlphabetic` is derived from categories and properties, so it might actually be possible to run that in reverse, that is we check `isAlphabetic` and excluding the others.